src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -149,8 +160,16 @@ BlockIO InterpreterDropQuery::executeToTableImpl(ASTDropQuery & query, DatabaseP
 
         if (query.kind == ASTDropQuery::Kind::Detach)
         {
-            getContext()->checkAccess(table->isView() ? AccessType::DROP_VIEW : AccessType::DROP_TABLE, table_id);
-            table->checkTableCanBeDetached();
+            getContext()->checkAccess(drop_storage, table_id);
+
+            if (table->isDictionary())
+            {
+                if (!query.is_dictionary)
+                    table->checkTableCanBeDropped();
+            }
+            else
+                table->checkTableCanBeDropped();
```

Comment:
```suggestion
            if (!table->isDictionary() || !query.is_dictionary)
                table->checkTableCanBeDetached();
```

Reply:
Btw, why not to check in this case? What will happen if `table->isDictionary() == true`, but `query.is_dictionary == false` and vice versa?

Reply:
This code migrated from previous method that processed dictionaries. I added comment. We check that dictionary can be dropped with DROP TABLE query, only table dictionaries that created as Dictionary view using CREATE TABLE () ENGINE=Dictionary, can be dropped with DROP TABLE.